src/Microsoft.Extensions.Logging.Filter/Internal/FilterLoggerFactory.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Extensions.Logging.Filter.Internal
+{
+    public class FilterLoggerFactory : ILoggerFactory
+    {
+        private readonly ILoggerFactory _innerLoggerFactory;
+        private readonly IFilterLoggerSettings _settings;
+
+        public FilterLoggerFactory(ILoggerFactory innerLoggerFactory, IFilterLoggerSettings settings)
+        {
+            _innerLoggerFactory = innerLoggerFactory;
+            _settings = settings;
+        }
+
+        public void AddProvider(ILoggerProvider provider)
+        {
+            var wrappedProvider = new FilterLoggerProvider(provider, _settings);
+            _innerLoggerFactory.AddProvider(wrappedProvider);
+        }
+
+        public ILogger CreateLogger(string categoryName)
+        {
+            throw new NotImplementedException();
```

Comment:
oh, goodness... this is very unfortunate... are we certain this shouldn't just call through? seems like it would be fairly frustrating to get a NotImplemented from CreateLogger, and learn that you need to "go up" to the original logger yourself...